test/orbit_determination.jl

```diff
@@ -0,0 +1,301 @@
+        idxs = findall(x -> x < 1, w8sveres17.(radec))
+        # Restricted Orbit Determination
+        sol = orbitdetermination(radec[idxs], params)
+
+        # Values by February 24, 2024
+
+        # Vector of observations
+        @test length(radec) == 883
+        @test numberofdays(radec) < 0.80
+        # Orbit solution
+        @test isa(sol, NEOSolution{Float64, Float64})
+        # Tracklets
+        @test length(sol.tracklets) == 2
+        @test sol.tracklets[1].radec[1] == radec[idxs[1]]
+        @test sol.tracklets[end].radec[end] == radec[idxs[end]]
+        @test issorted(sol.tracklets)
+        # Backward integration
+        @test datetime2days(date(radec[idxs[1]])) > sol.bwd.t0 + sol.bwd.t[end]
+        @test all( norm.(sol.bwd.x, Inf) .< 2 )
+        @test isempty(sol.t_bwd) && isempty(sol.x_bwd) && isempty(sol.g_bwd)
+        # Forward integration
+        @test datetime2days(date(radec[idxs[end]])) < sol.fwd.t0 + sol.fwd.t[end]
+        @test all( norm.(sol.fwd.x, Inf) .< 1e4 )
+        @test isempty(sol.t_fwd) && isempty(sol.x_fwd) && isempty(sol.g_fwd)
+        # Vector of residuals
+        @test length(sol.res) == 20
+        @test iszero(count(outlier.(sol.res)))
+        # Least squares fit
+        @test sol.fit.success
+        @test all( sigmas(sol) .< 2e-5 )
+        @test nrms(sol) < 0.30
+        # Scalig factors
+        @test all(sol.scalings .< 1e-5)
+        # Compatibility with JPL
+        JPL = [0.9741070119227359, 0.21515061351517384, 0.09390897837680391,
+               -0.007890445009307178, 0.016062726197198392, 0.006136042043681892]
+        @test all(abs.(sol() - JPL) ./ sigmas(sol) .< 0.3)
+
+        # Update orbit with more observations
+        sol1 = orbitdetermination(radec[1:30], sol, params)
+
+        # Orbit solution
+        @test isa(sol1, NEOSolution{Float64, Float64})
+        # Tracklets
+        @test length(sol1.tracklets) == 5
+        @test sol1.tracklets[1].radec[1] == radec[1]
+        @test sol1.tracklets[end].radec[end] == radec[30]
+        @test issorted(sol1.tracklets)
+        # Backward integration
+        @test datetime2days(date(radec[idxs[1]])) > sol1.bwd.t0 + sol1.bwd.t[end]
+        @test all( norm.(sol1.bwd.x, Inf) .< 1 )
+        @test isempty(sol1.t_bwd) && isempty(sol1.x_bwd) && isempty(sol1.g_bwd)
+        # Forward integration
+        @test datetime2days(date(radec[idxs[end]])) < sol1.fwd.t0 + sol1.fwd.t[end]
+        @test all( norm.(sol1.fwd.x, Inf) .< 1e4 )
+        @test isempty(sol1.t_fwd) && isempty(sol1.x_fwd) && isempty(sol1.g_fwd)
+        # Vector of residuals
+        @test length(sol1.res) == 30
+        @test iszero(count(outlier.(sol1.res)))
+        # Least squares fit
+        @test sol1.fit.success
+        @test all( sigmas(sol1) .< 2e-6 )
+        @test nrms(sol1) < 0.37
+        # Scalig factors
+        @test all(sol1.scalings .< 1e-6)
+        # Compatibility with JPL
+        @test all(abs.(sol1() - JPL) ./ sigmas(sol1) .< 1.6)
```

Comment:
JPL gives no information about which observations they discard as outliers or which weights they use; therefore, a strict comparison with JPL is not possible. Furthermore, the first orbit is well constrained because it is computed over the best quality astrometry, the observations added for the second orbit do not span a significant amount of extra time and may only introduce noise and/or outliers. Yet, the comparison stays within a 3 sigma threshold wrt JPL. This particular NEO requires a detailed filtering of the observations.

Reply:
Okay, if comparison wrt a JPL solution is not straightforward, then probably it makes sense to test for improvement of the post-fit statistics for each solution, e.g.:
```julia
@test nrms(sol1) < nrms(sol)
@test all.(sigmas(sol1) .< sigmas(sol))
```

Reply:
The NRMS increases from 0.29 to 0.37 from `sol` to `sol1` but `sigmas(sol1)` are smaller than `sigmas(sol)` by a factor of 4. I've added the test
```julia
@test all(sigmas(sol1) .< sigmas(sol))
```

Reply:
Alright, then I'd only suggest now to add the NRMS test commented for now and leave as a TODO to try to understand a bit better the differences wrt to the JPL solution.